src/Symfony/Component/Workflow/Registry.php

```diff
@@ -24,17 +23,11 @@ class Registry
     private $workflows = array();
 
     /**
-     * @param Workflow                        $workflow
-     * @param string|SupportStrategyInterface $supportStrategy
+     * @param Workflow                 $workflow
+     * @param SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $supportStrategy)
+    public function add(Workflow $workflow, SupportStrategyInterface $supportStrategy)
```

Comment:
@lyrixx:
To explain with more words:
Keep the `add(Workflow $workflow, $supportStrategy)` for 3.x, but starting from 3.4, you can add this code in the add function:

```php
public function add(Workflow $workflow, $supportStrategy)
{
    if (!$supportStrategy instanceof SupportStrategyInterface) {
        @trigger_error('Using a strategy that does not implement '.SupportStrategyInterface::class.' is deprecated since version 3.4 and will be removed in 4.0.', E_USER_DEPRECATED);
    }
}
```

This is for 3.4

And for 4.0, you remove this deprecation and add the type-hint for `SupportStrategyInterface`.

That's all.